modules/dbparser/radix.c

```diff
@@ -138,11 +138,37 @@ r_parser_pcre(gchar *str, gint *len, const gchar *param, gpointer state, RParser
 {
   RParserPCREState *self = (RParserPCREState *) state;
   gint rc;
-  gint matches[2];
+  gint num_matches;
 
-  rc = pcre_exec(self->re, self->extra, str, strlen(str), 0, 0, matches, 2);
-  if (rc <= 0)
-    return FALSE;
+  if (pcre_fullinfo(self->re, self->extra, PCRE_INFO_CAPTURECOUNT, &num_matches) < 0)
+    g_assert_not_reached();
+
+  if (num_matches > RE_MAX_MATCHES)
+    num_matches = RE_MAX_MATCHES;
+
+  gsize matches_size = 3 * (num_matches + 1);
+  gint *matches = g_alloca(matches_size * sizeof(gint));
+
+  rc = pcre_exec(self->re, self->extra, str, strlen(str), 0, 0, matches, matches_size);
```

Comment:
As far as I can see the matches are not *really* needed, but the length of matched string. Is there a way to only obtain that much ?
If not, the allocation part could be at least moved to the `r_parser_pcre_compile_state`.

Reply:
As far as I can see there is no way to only obtained the length of matched string (peeked inside the source).
Also if we do not provide - enough - space for matches, internally it is going to call for `pcre_malloc` (that user could override, but still), therefore I think it is better to allocate in the `r_parser_pcre_compile_state` as suggested before.

Reply:
I can agree with @Kokan that calculating the matches_size can be done in compile() step, next to `pcre_compile` and `pcre_study`.
It does not seem to be a big effort either.

However, then it should be done in `log_matcher_pcre` too, right? I wouldn't like to increase the scope of this PR, but it would only makes sense.

Reply:
I moved the alloc to compile!
I will open another PR for the `log_matcher_pcre` :)